### ISSUE TITLE
Code is running fine with linux but issue wrt the output from file student.py

### ISSUE THREAD

**Issue by gauravharjule** [DoctorKey/LSHFM.singleclassification#2]
1. I am now able to run the code for teacher and train it successfully for 240 epochs. 
2. Unable to get output while running the knowledge distillation code(train_student.py), as after the first epoch
I am getting the values as nan for losses, kd and cls.
3. I have tried debugging it but was unable to reach the output, request please guide for the same.
4. Also the losses when I go for cross architecture are very high, is here anything that I should init or I am missing

Regards
![image](https://github.com/DoctorKey/LSHFM.singleclassification/assets/129313849/5716b4b1-3151-490d-b7da-59b1ac8e53da)

**Comment by DoctorKey** [comment]
For the nan loss, you can try to tune the learning rate as well as the balance weights for the loss. In addition, you can train ResNet rather than VGG because it is more easier to train ResNet.

**Comment by gauravharjule** [comment]
1. Can you suggest a method to balance weights for the loss as well as the learning rate
2. The value of N taken here as hyper parameters is Dt*32, so should it also be changed?
3. When we train teacher and student before the KD,(python train_vanilla.py --model [teacher/student network] --gpus 0) it gets saved in the save folder for both teacher and student, but the  code given for knowledge distillation (train_student.py) pics the trained teacher from the above folder but student is picked from the models file here which is not pre-trained, contradicting as claimed in the paper it should also pic the trained student for KD when using fwg commmand:-
python train_student.py --model_s wrn_16_2 --path_t save/models/wrn_40_2_vanilla/ckpt_epoch_240.pth --gpus 0
4. Can you please guide for the same.

**Comment by gauravharjule** [comment]
hello 
I was able to run the code
Similarly was trying to implement LSH for cross model architecture but was unable to do so

Can you guide me with the same
did you attemt that approach and is it worth going forward or there are issues
